base/commands/viridian/viridian_cluster_create.go

```diff
@@ -0,0 +1,105 @@
+package viridian
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/output"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/viridian"
+)
+
+type ClusterCreateCmd struct{}
+
+func (cm ClusterCreateCmd) Init(cc plug.InitContext) error {
+	cc.SetCommandUsage("create-cluster [flags]")
+	long := `Creates a Viridian cluster for the logged in API key.
+
+Make sure you login before running this command.
+`
+	short := "Creates Viridian cluster"
+	cc.SetCommandHelp(long, short)
+	cc.SetPositionalArgCount(0, 0)
+	cc.AddStringFlag(propAPIKey, "", "", false, "Viridian API Key")
+	cc.AddStringFlag(flagName, "", "", false, "override the cluster name")
+	cc.AddStringFlag(flagPlan, "", "", false, "plan for the cluster, supported values: serverless")
+	cc.AddBoolFlag(flagDevelopment, "", false, false, "start a development cluster")
+	return nil
+}
+
+func (cm ClusterCreateCmd) Exec(ctx context.Context, ec plug.ExecContext) error {
+	api, err := getAPI(ec)
+	if err != nil {
+		return err
+	}
+	name := ec.Props().GetString(flagName)
+	plan := ec.Props().GetString(flagPlan)
+	if err := validatePlan(plan); err != nil {
+		return err
+	}
+	dev := ec.Props().GetBool(flagDevelopment)
+	csi, stop, err := ec.ExecuteBlocking(ctx, func(ctx context.Context, sp clc.Spinner) (any, error) {
+		sp.SetText("Retrieving available Kubernetes clusters")
```

Comment:
Let's not leak this information, moving `sp.SetText("Creating a cluster")` here would work better.

Reply:
We also have `Creating the cluster` below, maybe I can delete this part?

Reply:
Yes, let's move that here.